### ISSUE TITLE
fs.stat fails without error intermittently

### ISSUE THREAD

**Issue by wvanjaarsveld** [pillarjs/send#95]
I'm running a test environment (ubuntu 14.04 LTS guest on Windows server host using Hyper-V). I'm using express static file serving which fails more often than not. When it fails it is because fs.stat goes to "no mans land". Sometimes it returns with a result but more often than not it doesn't fire the callback at all.

```javascript
  SendStream.prototype.sendFile = function sendFile(path) {
  var i = 0
  var self = this

  debug('stat "%s"', path);
  fs.stat(path, function onstat(err, stat) {  // CALLBACK DOESN'T FIRE
    if (err && err.code === 'ENOENT'
      && !extname(path)
      && path[path.length - 1] !== sep) {
```

Have you ever come across anything like this because I'm not sure where to go next with this.

**Comment by dougwilson** [comment]
Hi! Unfortunately I have never seen such an issue before. If `fs.stat` never returns, there really isn't anything this module can do, unless you have a suggestion? Have you opened an issue with Node.js, which provides the `fs.stat` function?

**Comment by dougwilson** [comment]
Node.js issues can be seen/opened at https://github.com/nodejs/node/issues

**Comment by dougwilson** [comment]
I may be able to take a look as well, to confirm if there is something we can do vs. it being a Node.js issue. If you are interested, can you provide the following information?

1. The version of Node.js you are using.
2. The version of this module you are using.
3. An entire application I can run locally and reproduces the issue. Try to use no other modules besides this one to try and reduce the chance that something else is introducing the problem. If you do include other module, please provide the exact versions of any that are used so I can install the same versions.
4. The exact steps to take to reproduce the issue.

**Comment by wvanjaarsveld** [comment]
Thanks for the reply! 

I took a chance seeing I traced it to the sending of the file. The node version is 4.2.4 however I now reverted to 0.12.7 and that worked fine again. My gut says it is the virtual setup in combination with the Node.js. I subsequently noticed other file based issues. Unfortunately I do not have time for further investigations at the moment so I will be reverting to 0.12.7 for the time being until I am graced with more time.

Thanks again for the response. Much appreciated!
